### PR TITLE
Add doc updates

### PR DESCRIPTION
**Pull request by oliviassss**
### Issue

https://github.com/kubernetes-sigs/aws-load-balancer-controller/issues/3340
https://github.com/kubernetes-sigs/aws-load-balancer-controller/issues/3333
https://github.com/kubernetes-sigs/aws-load-balancer-controller/issues/3259
https://github.com/kubernetes-sigs/aws-load-balancer-controller/issues/3240

### Description
Doc updates for:
- Update legacy provider section with in-tree controller
- Add a pathType note in Ingress Spec
- Update NLB security group section
- Add compatibility matrix of k8s version in helm chart README

Checked the rendering via `make docs-preview`.

### Checklist
- [ ] Added tests that cover your change (if possible)
- [ ] Added/modified documentation as required (such as the `README.md`, or the `docs` directory)
- [ ] Manually tested
- [ ] Made sure the title of the PR is a good description that can go into the release notes

### BONUS POINTS checklist: complete for good vibes and maybe prizes?! :exploding_head:
- [ ] Backfilled missing tests for code in same general area :tada:
- [ ] Refactored something and made the world a better place :star2:

### CODE REVIEW

File: docs/guide/ingress/spec.md [hunk 1]
```diff
@@ -1,6 +1,13 @@
 # Ingress specification
 This document covers how ingress resources work in relation to The AWS Load Balancer Controller.
 
+!!!note ""
+    - Beginning from v2.4.3 of the AWS LBC, rules of `pathType: ImplementationSpecific` are always ordered according to the rule order in the manifest.
+    - However, in case of mixed path types,  
+        - `pathType: Exact` match are always ordered first
```
Comment (johngmyers) [inline docs/guide/ingress/spec.md:7]:
```suggestion
        - `pathType: Exact` paths are always ordered first 
```

File: docs/guide/ingress/spec.md [hunk 2]
```diff
@@ -1,6 +1,13 @@
 # Ingress specification
 This document covers how ingress resources work in relation to The AWS Load Balancer Controller.
 
+!!!note ""
+    - Beginning from v2.4.3 of the AWS LBC, rules of `pathType: ImplementationSpecific` are always ordered according to the rule order in the manifest.
+    - However, in case of mixed path types,  
+        - `pathType: Exact` match are always ordered first 
+        - followed by `pathType: Prefix` paths with the longest prefix first
```
Comment (johngmyers) [inline docs/guide/ingress/spec.md:8]:
```suggestion
        - followed by `pathType: Prefix` paths, with the longest prefix first
```

File: docs/guide/ingress/spec.md [hunk 3]
```diff
@@ -1,6 +1,13 @@
...
+        - `pathType: ImplementationSpecific` paths in the order from the manifest
```
Comment (johngmyers) [inline docs/guide/ingress/spec.md:9]:
```suggestion
        - followed by `pathType: ImplementationSpecific` paths, in the order they are listed in the manifest
```

File: docs/guide/ingress/spec.md [hunk 4]
```diff
@@ -1,6 +1,13 @@
 # Ingress specification
 This document covers how ingress resources work in relation to The AWS Load Balancer Controller.
 
+!!!note ""
+    - Beginning from v2.4.3 of the AWS LBC, rules of `pathType: ImplementationSpecific` are always ordered according to the rule order in the manifest.
+    - However, in case of mixed path types,
```
Comment (johngmyers) [inline docs/guide/ingress/spec.md:6]: The second part of this sentence is redundant with the second list item:

```suggestion
    Beginning from v2.4.3 of the AWS LBC, rules are ordered as follows:
```

File: docs/guide/service/annotations.md [hunk 5]
```diff
@@ -480,6 +480,10 @@ Load balancer access can be controlled via following annotations:
...
+The AWS Load Balancer Controller manages Kubernetes Services in a compatible way with the legacy aws cloud provider.
 
-The legacy cloud provider patch was added in Kubernetes v1.20 and is backported to Kubernetes v1.18.18+, v1.19.10+.
+- For users on v2.5.0+, The AWS LBC provides a mutating webhook for service resources to set the `spec.loadBalancerCLass` field for the serive of type LoadBalancer, which makes the AWS LBC the default controller for service of type LoadBalancer. 
+  Users can disable this feature and revert to set Cloud Controller Manager (in-tree controller) as the default by setting the helm chart value enableServiceMutatorWebhook to false with `--set enableServiceMutatorWebhook=false` .
```
Comment (johngmyers) [inline docs/guide/service/annotations.md:486]:
```suggestion
  Users can disable this feature and revert to using the AWS Cloud Controller Manager as the default service controller by setting the helm chart `enableServiceMutatorWebhook` value to false with `--set enableServiceMutatorWebhook=false` .
```

File: docs/guide/service/nlb.md [hunk 6]
```diff
@@ -161,7 +161,8 @@ service.beta.kubernetes.io/aws-load-balancer-proxy-protocol: "*"
 See [Subnet Discovery](../../deploy/subnet_discovery.md) for details on configuring Elastic Load Balancing for public or private placement.
 
 ## Security group
-AWS doesn't support attaching security groups to NLBs. To allow inbound traffic from an NLB, the controller automatically adds inbound rules to the worker node security groups, by default.
+ - From v2.6.0, the AWS LBC creates and attaches frontend and backend security groups to NLB by default. For more information please check [here](../../deploy/security_groups.md)
```
Comment (johngmyers) [inline docs/guide/service/nlb.md:164]:
```suggestion
 - From v2.6.0, the AWS LBC creates and attaches frontend and backend security groups to NLB by default. For more information please see [the security groups documentation](../../deploy/security_groups.md).
```

**Review by johngmyers — CHANGES_REQUESTED** [review]

File: docs/guide/service/annotations.md [hunk 7]
```diff
@@ -480,6 +480,10 @@ Load balancer access can be controlled via following annotations:
...
+- For users on older versions, the annotation `service.beta.kubernetes.io/aws-load-balancer-type` is used to determine which controller reconciles the service. If the annotation value is `nlb-ip` or `external`, 
+  legacy cloud provider ignores the service resource (provided it has the correct patch) so that the AWS LBC can take over. For all other values of the annotation, the legacy cloud provider will handle the service. 
+  Note that this annotation should be specified during service creation and not edited later. The legacy cloud provider patch was added in Kubernetes v1.20 and is backported to Kubernetes v1.18.18+, v1.19.10+.
```
[the remark this answers is not in the document]
Comment (oliviassss) [reply docs/guide/service/annotations.md:489]: Adapted with minor change.

**Pull request merged**